esp/services/ws_dfu/ws_dfuService.cpp

```diff
@@ -6153,32 +6022,86 @@ SecAccessFlags translateToSecAccessFlags(CSecAccessType from)
     }
 }
 
+void CWsDfuEx::dFULookupFileCommon(IEspContext &context, const CDfsLogicalFileName &lfn, const char *requestId, unsigned expirySecs, bool returnTextResponse, IEspDFUFileAccessInfo &accessInfo)
+{
+    StringBuffer fileName;
+    lfn.get(fileName, false, true);
+    if (0 == fileName.length())
+         throw MakeStringException(ECLWATCH_INVALID_INPUT, "No Name defined.");
+
+    StringBuffer userID;
+    context.getUserID(userID);
+
+    Owned<IUserDescriptor> userDesc;
+    if (!userID.isEmpty())
+    {
+        userDesc.setown(createUserDescriptor());
+        userDesc->set(userID.str(), context.queryPassword(), context.querySignature());
+    }
+
+    checkLogicalName(fileName, userDesc, true, false, false, nullptr); // check for read permissions
+
+    Owned<IDistributedFile> df = queryDistributedFileDirectory().lookup(fileName, userDesc, false, false, true); // lock super-owners
+    if (!df)
+        throw MakeStringException(ECLWATCH_FILE_NOT_EXIST,"Cannot find file '%s'.", fileName.str());
+
+    StringBuffer cluster;
+    lfn.getCluster(cluster);
+    Owned<IFileDescriptor> fileDesc = df->getFileDescriptor(cluster);
+
+    // NB: if file has copies on >1 cluster, they must share the same key
+
+    std::vector<std::string> groups;
+    unsigned numClusters = df->numClusters();
+    for (unsigned c=0; c<numClusters; c++)
+    {
+        StringBuffer clusterName;
+        const char *clusterGroup = df->getClusterName(c, clusterName.clear());
+        groups.push_back(clusterGroup);
+    }
+
+    StringBuffer keyPairName;
+    unsigned port;
+    bool secure;
+    getFileDafilesrvConfiguration(keyPairName, port, secure, fileName, groups);
+
+    Owned<IPropertyTree> metaInfo = createDFUFileMetaInfo(fileName, fileDesc, requestId, "READ", expirySecs, userDesc, keyPairName, port, secure, maxFileAccessExpirySeconds);
```

Comment:
Is it worthwhile to create an enum for the accesstype (I thought one might already exist to be honest) to avoid stringwise operations on the "READ" and/or "WRITE" ?

Reply:
not sure, there's no string comparison until it hits the server (dafilesrv).
i.e. "READ" is in human readable form in the JSON request. 
Where possible, and cost is inconsequential, I think it's best that the JSON request fields remains readable, vs numbers matching enums in a header file somewhere.